### PR TITLE
Support wide output format for get commands (-o wide)

### LINKED ISSUE #2889: Support wide output format for get commands (`-o wide`)
Refer https://gateway-api.sigs.k8s.io/geps/gep-2722/ for details

### PR DESCRIPTION
**Pull request by yeedove**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2889

**Does this PR introduce a user-facing change?**:

```release-note
1.Support wide output format for get commands (-o wide)
2.Replace `Print` in BackendsPrinter with `PrintTable`
```

```yaml
cat <<EOF | kubectl apply -f -
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: BackendTLSPolicy
metadata:
  name: enable-backend-tls
  namespace: default
spec:
  targetRef:
    group: ''
    kind: Service
    name: tls-backend
    sectionName: "443"
  tls:
    caCertRefs:
    - name: example-ca
      group: ''
      kind: ConfigMap
    hostname: www.example.com
EOF

```
- **backend**
```
➜  ✗ gwctl get backend       
NAMESPACE  NAME         TYPE     AGE
default    backend      Service  106d
default    kubernetes   Service  106d
default    tls-backend  Service  44h

➜  ✗ gwctl get backend -owide
NAMESPACE  NAME         TYPE     AGE   REFERRED BY ROUTES  POLICIES
default    backend      Service  106d  None                0
default    kubernetes   Service  106d  None                0
default    tls-backend  Service  44h   default/backend     1
```

- **gatewayclass**
```
➜  ✗ gwctl get gatewayclass       
NAME           CONTROLLER                                     ACCEPTED  AGE
eg             gateway.envoyproxy.io/gatewayclass-controller  True      104d
envoy-gateway  gateway.envoyproxy.io/gatewayclass-controller  True      95d

➜  ✗ gwctl get gatewayclass -owide
NAME           CONTROLLER                                     ACCEPTED  AGE   GATEWAYS
eg             gateway.envoyproxy.io/gatewayclass-controller  True      104d  1
envoy-gateway  gateway.envoyproxy.io/gatewayclass-controller  True      95d   0
```

- **gateway**
```
➜  ✗ gwctl get gateway      
NAMESPACE  NAME  CLASS  ADDRESSES       PORTS  PROGRAMMED  AGE
default    eg    eg     172.18.255.201  80     True        106d

➜  ✗ gwctl get gateway -owide
NAMESPACE  NAME  CLASS  ADDRESSES       PORTS  PROGRAMMED  AGE   POLICIES  HTTPROUTES
default    eg    eg     172.18.255.201  80     True        106d  0         1
```

- **httproute**
```
➜  ✗ gwctl get httproute
NAMESPACE  NAME     HOSTNAMES        PARENT REFS  AGE
default    backend  www.example.com  1            104d
➜  ✗ gwctl get httproute -owide
NAMESPACE  NAME     HOSTNAMES        PARENT REFS  AGE   POLICIES
default    backend  www.example.com  1            104d  0

```

- **namespace**
```
➜  ✗ gwctl get ns
NAME                  STATUS  AGE
default               Active  104d
envoy-gateway-system  Active  104d
kube-node-lease       Active  104d
kube-public           Active  104d
kube-system           Active  104d
local-path-storage    Active  104d
metallb-system        Active  95d
monitoring            Active  94d
➜  ✗ gwctl get ns -owide
NAME                  STATUS  AGE   POLICIES
default               Active  104d  0
envoy-gateway-system  Active  104d  0
kube-node-lease       Active  104d  0
kube-public           Active  104d  0
kube-system           Active  104d  0
local-path-storage    Active  104d  0
metallb-system        Active  95d   0
monitoring            Active  94d   0
```

### CODE REVIEW

File: gwctl/pkg/utils/types.go [hunk 1]
```diff
@@ -100,13 +100,16 @@ func MustPolicyManagerForTest(t *testing.T, fakeClients *common.K8sClients) *pol
 type OutputFormat string
 
 const (
+	OutputFormatWIDE  OutputFormat = "wide"
```
Comment (mlavacca) [inline gwctl/pkg/utils/types.go:103]:
```suggestion
	OutputFormatWide  OutputFormat = "wide"
```
As wide is not an acronym for something but an actual word, we should keep it `CamelCase`.

**Review by mlavacca — COMMENTED** [review]
This looks good to me, I left a minor comment 👍 

/ok-to-test

**Review by jongwooo — COMMENTED** [review]
The `OutputFormatWIDE` type is not currently handled by the switch statement below. Please add a case statement for `OutputFormatWIDE` type:

https://github.com/kubernetes-sigs/gateway-api/blob/e50fd3b065d16bb5c4f6462a0a8dfbfccdd5730e/gwctl/pkg/printer/policies.go#L88-L96

https://github.com/kubernetes-sigs/gateway-api/blob/e50fd3b065d16bb5c4f6462a0a8dfbfccdd5730e/gwctl/pkg/printer/policies.go#L129-L137

**Review by mlavacca — COMMENTED** [review]
Thanks, @yeedove 

/lgtm

I'll let the approval to @gauravkghildiyal

File: gwctl/pkg/printer/backends.go [hunk 2]
```diff
@@ -34,9 +34,20 @@ type BackendsPrinter struct {
...
+func (bp *BackendsPrinter) GetPrintableNodes(resourceModel *resourcediscovery.ResourceModel) []NodeResource {
+	return NodeResources(maps.Values(resourceModel.GatewayClasses))
+}
+
+func (bp *BackendsPrinter) PrintTable(resourceModel *resourcediscovery.ResourceModel, wide bool) {
+	var columnNames []string
+	if wide {
+		columnNames = []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE", "POLICIES"}
+	} else {
+		columnNames = []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE"}
```
Comment (gauravkghildiyal) [inline gwctl/pkg/printer/backends.go:46]: I know this is a bit different from what's documented but I've come to realize that we should avoid any parameter from being displayed which needs additional calculations. 

Calculating things like "REFERRED BY ROUTES" and "POLICIES" will take additional time (which becomes significant when there's lot's of resources). Hence it's preferrable to only include them in the `-o wide` format and not as a default.

tl;dr: Can you please move `REFERRED BY ROUTES` to wide format as well?

(In case you are wondering, we'll separately work on not calculating the additional values by default int the resourceModel -- as of now, that distinction doesn't exist`

**Review by gauravkghildiyal — COMMENTED** [review]
Thanks @yeedove! This looks great, just have one minor request which should be quite straightforward.

**Comment by gauravkghildiyal** [comment]
Can you please also edit the release note and avoid the `BackendsPrinter` (to only focus on how the user experience is different)

File: gwctl/pkg/printer/backends.go (hunk 2 above)
Comment (yeedove) [reply gwctl/pkg/printer/backends.go]: Of course, done!
